hashicorp/terraform-provider-tfe
# Allow team and organization_membership to be imported by user facing values

## Description

Allow the tfe_team resource to be imported by `<org name>/<team name>`
Allow the tfe_team_organization_member resources to be imported by `<org-name>/<team name>/<user-email>`

Presently, both these resources can only be imported through their ID, which is are presented to the user in the TFE UI. 

When importing a team, we will now check if the relevant peice of the import ID matches the format of a TFE resource ID. If not, or if no team resource exists with that ID, we can search for teams with that name in the given organization

## Testing plan

1.  Check that tfe_team can still be imported by the existing ID
2. Check that tfe_team_org_member can still be imported by the existing ID
3. Check that tfe_team can now be imported by team name
4. Check that tfe_team_org_member can now be imported by org and team name + user email
5. Check that invalid ID formats exist gracefully for both resources
6. Check that importing by a non-existent org name errors gracefully for both resources
7. Check that importing by a non-existent team name errors gracefully for both resources
8. Check that importing team membership by a user email which does not belong to the org fails gracefully

## External links

- [JIRA for team](https://hashicorp.atlassian.net/browse/TF-3356)
- [JIRA for org team memeber](https://hashicorp.atlassian.net/browse/TF-3336)


## Output from acceptance tests

```
❯ go test ./...  -run TestAccTFETeam_ -tags=integration -v
?       github.com/hashicorp/terraform-provider-tfe     [no test files]
=== RUN   TestAccTFETeam_basic
--- PASS: TestAccTFETeam_basic (8.00s)
=== RUN   TestAccTFETeam_full
--- PASS: TestAccTFETeam_full (7.72s)
=== RUN   TestAccTFETeam_full_update
--- PASS: TestAccTFETeam_full_update (17.36s)
=== RUN   TestAccTFETeam_import_byId
--- PASS: TestAccTFETeam_import_byId (8.52s)
=== RUN   TestAccTFETeam_import_byId_doesNotExist
--- PASS: TestAccTFETeam_import_byId_doesNotExist (8.50s)
=== RUN   TestAccTFETeam_import_byName
--- PASS: TestAccTFETeam_import_byName (8.77s)
=== RUN   TestAccTFETeam_import_missingOrg
--- PASS: TestAccTFETeam_import_missingOrg (9.43s)
=== RUN   TestAccTFETeam_import_missingTeam
--- PASS: TestAccTFETeam_import_missingTeam (8.37s)
=== RUN   TestAccTFETeam_import_teamNameWithSpaces
--- PASS: TestAccTFETeam_import_teamNameWithSpaces (8.62s)
=== RUN   TestAccTFETeam_import_teamNameWithSlashes
--- PASS: TestAccTFETeam_import_teamNameWithSlashes (8.50s)
=== RUN   TestAccTFETeam_import_teamNameWhichLooksLikeID
--- PASS: TestAccTFETeam_import_teamNameWhichLooksLikeID (8.53s)
PASS
ok      github.com/hashicorp/terraform-provider-tfe/tfe 102.647s

❯ go test ./...  -run TestAccTFETeamOrganizationMember_ -tags=integration -v
?       github.com/hashicorp/terraform-provider-tfe     [no test files]
=== RUN   TestAccTFETeamOrganizationMember_basic
--- PASS: TestAccTFETeamOrganizationMember_basic (10.43s)
=== RUN   TestAccTFETeamOrganizationMember_import_byId
--- PASS: TestAccTFETeamOrganizationMember_import_byId (9.66s)
=== RUN   TestAccTFETeamOrganizationMember_import_byTeamName
--- PASS: TestAccTFETeamOrganizationMember_import_byTeamName (10.27s)
=== RUN   TestAccTFETeamOrganizationMember_import_orgDoesNotExist
--- PASS: TestAccTFETeamOrganizationMember_import_orgDoesNotExist (9.59s)
=== RUN   TestAccTFETeamOrganizationMember_import_teamNameDoesNotExist
--- PASS: TestAccTFETeamOrganizationMember_import_teamNameDoesNotExist (9.73s)
=== RUN   TestAccTFETeamOrganizationMember_import_userEmailDoesNotExist
--- PASS: TestAccTFETeamOrganizationMember_import_userEmailDoesNotExist (9.46s)
=== RUN   TestAccTFETeamOrganizationMember_import_incorrectFormat
--- PASS: TestAccTFETeamOrganizationMember_import_incorrectFormat (10.15s)
PASS
ok      github.com/hashicorp/terraform-provider-tfe/tfe 69.605s


...
```
